### PR TITLE
Compose initial pooltogether governor

### LINKED ISSUE #1: Assemble the PoolTogether Governor
Create the PoolTogether Governor using appropriate OpenZeppelin Governor modules and the FlexibleVoting extension.

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Leverage Gitcoin governor upgrade to compose the correct governor for pooltogether
- There are no tests or deploy scripts those will be handled in subsequent prs.
- Add License
- Bump down coverage

Closes #1

### CODE REVIEW

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/4" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/4?utm_source=stack-comment).

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [inline src/PooltogetherGovernor.sol:75]: This was added on top of the Gitcoin upgrade

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [inline src/PooltogetherGovernor.sol:16]: This was pretty much all copied from the Gitcoin upgrade

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [inline src/PooltogetherGovernor.sol:28]: How concerned are we if some methods are not present on the token contract? One example would be `getVotes` is missing, but there is a replacement `getCurrentVotes`.

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [inline src/PooltogetherGovernor.sol:34]: Same here some of the method names are different from the interface. An example is the interface has `GRACE_PERIOD` and the pooltogether timelock has `gracePeriod`.

File: src/PooltogetherGovernor.sol
Comment (apbendi) [reply src/PooltogetherGovernor.sol:28]: The big thing we'll need to understand is whether the existing contract ABI's play nicely with what the OZ extensions are expecting, or if they are breaking changes that will require us to modify the OZ contracts in some way. I can help dig into this—it should also become apparent once we start writing tests, which would fail if the signatures didn't work.

File: src/PooltogetherGovernor.sol
Comment (apbendi) [reply src/PooltogetherGovernor.sol:34]: Same comment as above—the question is whether this will cause breakage with the OZ contract out of the box. If it does, we'll have to modify the OZ contracts to account for the differences.

File: src/PooltogetherGovernor.sol
Comment (apbendi) [reply src/PooltogetherGovernor.sol:75]: Hmm interesting, not totally clear to me why that is, but we can investigate.

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [reply src/PooltogetherGovernor.sol:28]: Ah okay, let's cross that bridge once we get there then. #3 for reference

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [reply src/PooltogetherGovernor.sol:34]: Tagging #3 for reference

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [reply src/PooltogetherGovernor.sol:138]: I think we are safe to ignore this right?

File: src/PooltogetherGovernor.sol
Comment (alexkeating) [reply src/PooltogetherGovernor.sol:75]: The error without this indicated it was implemented by both the `Governor` and `GovernorCountingFractional` contacts.

File: src/PooltogetherGovernor.sol
Comment (apbendi) [reply src/PooltogetherGovernor.sol:28]: I think we should investigate this deeper as a standalone task, but agree it can wait until after this PR is merged. https://github.com/ScopeLift/pooltogether-governor-upgrade/issues/5

File: src/PooltogetherGovernor.sol
Comment (apbendi) [reply src/PooltogetherGovernor.sol:75]: I see why we're getting this now and it's not a problem 👍

**Review by apbendi — APPROVED** [review]
Good to go, thanks @alexkeating.

**Pull request merged**
